presto-native-execution/presto_cpp/main/operators/tests/UnsafeRowShuffleTest.cpp

```diff
@@ -539,12 +552,86 @@ TEST_F(UnsafeRowShuffleTest, endToEnd) {
       std::string(TestShuffleFactory::kShuffleName),
       kShuffleInfo,
       kShuffleInfo,
+      false,
       numPartitions,
       numMapDrivers,
       {data});
   TestShuffleWriter::reset();
 }
 
+TEST_F(UnsafeRowShuffleTest, replicateNullPartitionStrategy) {
+  auto data = vectorMaker_.rowVector({
+      makeNullableFlatVector<int32_t>({1, 2, std::nullopt, std::nullopt, 4}),
+      makeNullableFlatVector<int64_t>({10, 20, std::nullopt, 50, std::nullopt}),
+  });
+  auto expectedData = vectorMaker_.rowVector({
+      makeNullableFlatVector<int32_t>(
+          {1,
+           2,
+           std::nullopt,
+           std::nullopt,
+           4,
+           std::nullopt, // Replicated Null for partition 0
+           std::nullopt, // Replicated Null for partition 0
+           std::nullopt, // Replicated Null for partition 2
+           std::nullopt}), // Replicated Null for partition 2
+      makeNullableFlatVector<int64_t>(
+          {10,
+           20,
+           std::nullopt,
+           50,
+           std::nullopt,
+           std::nullopt, // Replicated Null for partition 0
+           50, // Replicated Null for partition 1
+           std::nullopt, // Replicated Null for partition 2
+           50}), // Replicated Null for partition 2
+  });
+
+  auto plan = exec::test::PlanBuilder()
+                  .values({data}, true)
+                  .addNode(addPartitionAndSerializeNode(3, true))
+                  .planNode();
+
+  testPartitionAndSerialize(plan, expectedData);
+}
+TEST_F(UnsafeRowShuffleTest, replicateAnyPartitionStrategy) {
+  auto data = vectorMaker_.rowVector({
+      makeNullableFlatVector<int32_t>({1, 2, 3, 4}),
+      makeNullableFlatVector<int64_t>({10, 20, 30, 40}),
+  });
+  auto expectedData = vectorMaker_.rowVector({
+      makeNullableFlatVector<int32_t>({1, 2, 3, 4, 1, 1, 1, 1}),
+      makeNullableFlatVector<int64_t>({10, 20, 30, 40, 10, 10, 10, 10}),
+  });
+
+  auto plan = exec::test::PlanBuilder()
+                  .values({data}, true)
+                  .addNode(addPartitionAndSerializeNode(7, true))
+                  .planNode();
+
+  testPartitionAndSerialize(plan, expectedData);
+}
+
+TEST_F(UnsafeRowShuffleTest, replicateNullAnyPartitionStrategy) {
+  auto data = vectorMaker_.rowVector({
+      makeNullableFlatVector<int32_t>({1, 2, std::nullopt}),
+      makeNullableFlatVector<int64_t>({10, 20, std::nullopt}),
+  });
+  auto expectedData = vectorMaker_.rowVector({
+      makeNullableFlatVector<int32_t>(
+          {1, 2, std::nullopt, std::nullopt, std::nullopt, std::nullopt}),
+      makeNullableFlatVector<int64_t>(
+          {10, 20, std::nullopt, std::nullopt, std::nullopt, std::nullopt}),
+  });
+
+  auto plan = exec::test::PlanBuilder()
+                  .values({data}, true)
+                  .addNode(addPartitionAndSerializeNode(4, true))
+                  .planNode();
+
+  testPartitionAndSerialize(plan, expectedData);
+}
+
```

Comment:
Can you add a fuzzer test to generate input using fuzzer vector and rely on the test utility I mentioned to verify the result. Mostly about the extra replicated rows. The test utility can take three inputs: (1) input vector; (2) output vector; (3) partition function? Thanks!

Reply:
@xiaoxmeng There was already a fuzzer test, I refactored that a bit to also test for replicateNullAndAny strategy.
It does not have all the 3 inputs, like you suggested however it still tests for replicateNullAndAny specifically. For the fuzzer test, I don't think we should assert with a output vector, since the output vector (with replicateNullAndAny enabled) isn't deterministic and can change with implementation.
However to test for replicateNullAndAny strategy, I have added asserts to test for null count and any value.